qiskit/synthesis/evolution/lie_trotter.py

```diff
@@ -79,8 +80,19 @@ def __init__(
                 built.
             wrap: Whether to wrap the atomic evolutions into custom gate objects. This only takes
                 effect when ``atomic_evolution is None``.
+            preserve_order: If ``False``, allows reordering the terms of the operator to
+                potentially yield a shallower evolution circuit. Not relevant
+                when synthesizing operator with a single term.
```

Comment:
I think from a docs perspective it would be good to explain the tradeoffs here, especially around runtime performance.